tests/kernel/critical/src/main.c

```diff
@@ -42,7 +42,12 @@ static struct k_work_q offload_work_q;
 static K_THREAD_STACK_DEFINE(offload_work_q_stack,
 			     CONFIG_OFFLOAD_WORKQUEUE_STACK_SIZE);
 
+#ifdef CONFIG_COVERAGE_GCOV
+#define STACK_SIZE 4096
+#else
 #define STACK_SIZE 1024
```

Comment:
just add CONFIG_TEST_EXTRA_STACKSIZE instead

Reply:
Ok, I will do the change. Please help review another pr for qemu_x86, two platforms failed on different cases.

Reply:
it's the same bug on both targets, just do in a single PR.